realm/realm-library/src/main/cpp/CMake/RealmCore.cmake

```diff
@@ -0,0 +1,125 @@
+#
+# Copyright 2017 Realm Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+###########################################################################
+include(ExternalProject)
+
+function(build_existing_realm_core core_source_path)
+    if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+        set(debug_lib_suffix "-dbg")
+        add_compile_options(-DREALM_DEBUG)
+    else()
+        add_compile_options(-DNDEBUG)
+    endif()
+
+    ExternalProject_Add(realm-core
+        SOURCE_DIR ${core_source_path}
+        PREFIX ${core_source_path}/build-android-${ANDROID_ABI}-${CMAKE_BUILD_TYPE}
+        CMAKE_ARGS  -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}
+                    -DANDROID_ABI=${ANDROID_ABI}
+                    -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+                    -DREALM_BUILD_LIB_ONLY=YES
+                    -DREALM_ENABLE_ENCRYPTION=1
+        INSTALL_COMMAND ""
+        LOG_CONFIGURE 1
+        LOG_BUILD 1
+        )
+
+    ExternalProject_Get_Property(realm-core SOURCE_DIR)
+    ExternalProject_Get_Property(realm-core BINARY_DIR)
+
+    # Create directories that are included in INTERFACE_INCLUDE_DIRECTORIES, as CMake requires they exist at
+    # configure time, when they'd otherwise not be created until we download and extract core.
+    file(MAKE_DIRECTORY "${BINARY_DIR}/src")
+
+    set(core_lib_file "${BINARY_DIR}/src/realm/librealm${debug_lib_suffix}.a")
+    add_library(lib_realm_core STATIC IMPORTED)
+    set_target_properties(lib_realm_core PROPERTIES IMPORTED_LOCATION ${core_lib_file}
+        IMPORTED_LINK_INTERFACE_LIBRARIES atomic
+        INTERFACE_INCLUDE_DIRECTORIES "${SOURCE_DIR}/src;${BINARY_DIR}/src")
+
+    ExternalProject_Add_Step(realm-core ensure-libraries
+        DEPENDEES build
+        BYPRODUCTS ${core_lib_file}
+        )
+
+    add_dependencies(lib_realm_core realm-core)
+endfunction()
+
+# Add the sync released as the library.
+function(use_sync_release enable_sync sync_dist_path)
+    # Link to core/sync debug lib for debug build if it is debug build and linking with debug core is enabled.
+    if (CMAKE_BUILD_TYPE STREQUAL "Debug" AND ${ENABLE_DEBUG_CORE})
+        set(debug_lib_suffix "-dbg")
+        add_compile_options(-DREALM_DEBUG)
+    else()
+        add_compile_options(-DNDEBUG)
+    endif()
+
+    # Configure import realm core lib
+    set(core_lib_path ${sync_dist_path}/librealm-android-${ANDROID_ABI}${debug_lib_suffix}.a)
+    if (NOT EXISTS ${core_lib_path})
+        if (ARMEABI)
```

Comment:
What about x86?

Reply:
core's ABI name for x86 is correct. This old code moved from `CMakeList.txt`. those thing could be removed later when sync switched to cmake.

Reply:
👍